synapse/federation/federation_client.py

```diff
@@ -593,7 +595,7 @@ async def make_membership_event(
             SynapseError: if the chosen remote server returns a 300/400 code, or
                 no servers successfully handle the request.
         """
-        valid_memberships = {Membership.JOIN, Membership.LEAVE}
+        valid_memberships = {Membership.JOIN, Membership.LEAVE, Membership.KNOCK}
```

Comment:
This is probably not worth fixing, but it is inconsistent that we always consider knock a valid membership type.

More importantly -- do we need to somewhere check the room version supports knocking?